backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/DemoUserIdentitySupplier.java

```diff
@@ -0,0 +1,60 @@
+package gov.cdc.usds.simplereport.config.authorization;
+
+import gov.cdc.usds.simplereport.config.simplereport.DemoUserConfiguration;
+import gov.cdc.usds.simplereport.config.simplereport.DemoUserConfiguration.DemoUser;
+import gov.cdc.usds.simplereport.service.model.IdentityAttributes;
+import gov.cdc.usds.simplereport.service.model.IdentitySupplier;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+/**
+ * An {@link IdentitySupplier} that looks up the username of the current authenticated (or, more
+ * likely, "authenticated") user in the list of configured demo users. It is like that had we set
+ * out to have this originally we would have ended up with a more conventional UserDetailsService,
+ * and we could eventually work our way back there, but path-dependence produces weird results
+ * sometimes.
+ */
+public class DemoUserIdentitySupplier implements IdentitySupplier {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DemoUserIdentitySupplier.class);
+
+  private Map<String, IdentityAttributes> _userLookup;
+  private IdentityAttributes _defaultIdentity;
+
+  public DemoUserIdentitySupplier(DemoUserConfiguration config) {
+    this(config.getAllUsers());
+    DemoUser defaultUser = config.getDefaultUser();
+    if (defaultUser != null) {
+      _defaultIdentity = defaultUser.getIdentity();
+    }
+  }
+
+  public DemoUserIdentitySupplier(Collection<DemoUser> wiredUsers) {
+    _userLookup =
+        wiredUsers.stream().collect(Collectors.toMap(DemoUser::getUsername, e -> e.getIdentity()));
+  }
+
+  @Override
+  public IdentityAttributes get() {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
do we need these first two if-checks? do we have compelling use cases for these in tests? if so, can we add comments to explain?

Reply:
No intentional use cases for them, no—but I'd rather have an `IllegalStateException` than a `NullPointerException` if we happen to find a way to trip that unintentionally.

Reply:
can you add a comment explaining that these two if-checks are just safeguards? it also might be clearer if for the `!auth.isAuthenticated()` case, you also threw an `IllegalStateException`

Reply:
As it turns out, I can't throw an exception in either case, since our CommandLineRunner that runs `initAll` doesn't run in a request context. They are both real cases for the demo-with-default-user case, so I will just note that in the appropriate place in #1001.

Reply:
Going to resolve this conversation as the code is already moving to a new location, and various assumptions changed on the way.